nemo/collections/nlp/data/language_modeling/megatron/t5_prompt_learning_dataset.py

```diff
@@ -84,9 +118,15 @@ def load_data(self, dataset):
             input_example = self._insert_virtual_token_placeholders(input_example, virtual_token_splits)
 
             # a trick to align with the data format in t5 pretraining
-            input_ids = self.tokenizer.text_to_ids(input_example) + self.tokenizer.text_to_ids(T5Sentinel.FIRST.value)
+            input_ids = self.tokenizer.text_to_ids(input_example)
+            if self.add_sentinel_to_input:
+                input_ids = input_ids + self.tokenizer.text_to_ids(T5Sentinel.FIRST.value)
 
             # Add BOS/EOS to the input of encoder if desired, adds EOS by default
+            if self.ul2_prompt_token is not None:
```

Comment:
Should probably add the `ul2_prompt_token` after the virtual prompt tokens if there are some at the beginning of the text input.